### PR TITLE
[c#] Dependency Pass

### LINKED ISSUE #4180: [c#] Dependency Pass of `.csproj`
It would be useful to understand which dependencies are in the program, especially for determining what type stubs we need to bring in.

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
Parse `.csproj` files to find dependencies.

Resolves #4180

### CODE REVIEW

File: joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  override def generateParts(): Array[File] = buildFiles.map(x => File(x)).toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, part: File): Unit = {
+    SecureXmlParsing.parseXml(part.contentAsString) match {
+      case Some(xml) if xml.label == "Project" =>
+        xml.child
+          .filter(_.label == "ItemGroup")
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala:23]: I guess these .filter/.flatmaps can be combined to .collect right?

File: joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          .flatMap { packageReference =>
+            Try {
+              val packageName    = packageReference.attribute("Include").map(_.toString()).get
+              val packageVersion = packageReference.attribute("Version").map(_.toString()).get
+              NewDependency()
+                .name(packageName)
+                .version(packageVersion)
+            }.toOption
+          }
+          .foreach(builder.addNode)
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala:35]: .addNode could also be called in the last .flatMap / .collect so we do not need an additional .foreach?

File: joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      case Some(_) =>
+      case None    => logger.error(s"Failed to parse build file ${part.pathAsString}")
+    }
+
+//    } match {
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala:40]: Leftover out-commented code.

**Review by max-leuthaeuser — APPROVED** [review]
Just some nits.

File: joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala (hunk 1 above)
Comment (DavidBakerEffendi) [reply joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/passes/DependencyPass.scala:23]: Seems I can `collect` but will still need to flatten `.child` but yeah would look neater

**Pull request merged**